INTDB-301-DOC: Move Federated Docs to separate branch (#747)
mongodb/terraform-provider-mongodbatlas: website/docs/r/federated_settings_org_role_mapping.html.markdown

---

# mongodbatlas_cloud_federated_settings_org_role_mapping

`mongodbatlas_cloud_federated_settings_org_role_mapping` provides an Role Mapping resource. This allows organization role mapping to be created.

## Example Usage

```terraform
resource "mongodbatlas_cloud_federated_settings_org_role_mapping" "org_group_role_mapping_import" {
  federation_settings_id = "627a9687f7f7f7f774de306f14"
  org_id                 = "627a9683e7f7f7ff7fe306f14"
  external_group_name    = "myGrouptest"

  role_assignments {
    org_id = "627a9683e7f7f7ff7fe306f14"
    roles     = ["ORG_MEMBER","ORG_GROUP_CREATOR","ORG_BILLING_ADMIN"]
  }

  role_assignments {
    group_id = "628aa20d7f7f7f7f7098b81b8"
    roles     = ["GROUP_OWNER","GROUP_DATA_ACCESS_ADMIN","GROUP_SEARCH_INDEX_EDITOR","GROUP_DATA_ACCESS_READ_ONLY"]
  }

  role_assignments {
    group_id = "628aa20d7f7f7f7f7078b81b8"
    roles     = ["GROUP_OWNER","GROUP_DATA_ACCESS_ADMIN","GROUP_SEARCH_INDEX_EDITOR","GROUP_DATA_ACCESS_READ_ONLY","GROUP_DATA_ACCESS_READ_WRITE"]
  }
}
```

## Argument Reference

* `federation_settings_id` - (Required) Unique 24-hexadecimal digit string that identifies the federated authentication configuration.

> * `org_id` - Unique 24-hexadecimal digit string that identifies the connected organization.

Reviewer: ```suggestion
* `org_id` - Unique 24-hexadecimal digit string that identifies the organization that contains your projects.
```

Reviewer 2: This specific instance was corrected, but is it correct to assume any other instance of `connected organization` should change to `organization that contains your projects`?

Reviewer: As noted in our standup I only changed the ones where this was different than the source docs at https://www.mongodb.com/docs/atlas/reference/api/federation-configuration/.  Both are technically correct so I'm trusting the docs are slightly different depending on the context (which seems to be the case)

* `external_group_name` - Unique human-readable label that identifies the identity provider group to which this role mapping applies.
* `role_assignments` - Atlas roles and the unique identifiers of the groups and organizations associated with each role.

> * `group_id` - Unique identifier of the project that owns this Role Mapping Configuration.

Reviewer: ```suggestion
* `group_id` - Unique identifier of the project to which you want the role mapping to apply
```

* `roles` - Specifies the Roles that are attached to the Role Mapping.

## Attributes Reference

In addition to all arguments above, the following attributes are exported:
* `id` - Unique 24-hexadecimal digit string that identifies this role mapping.

## Import

> FederatedSettingsOrgRoleMapping must be imported using federation_settings_id-org_id-role_mapping_id, e.g.

Reviewer: Does this have to be imported if there are no role mappings yet?  I think you can setup an IdP without a role mapping (hence this also being a POST).  Also if it does require require it again, should be called out at the top.

Reviewer: Still open - I don't think this one must be.
